client-course-schedulizer/src/readCSV.ts

```diff
@@ -0,0 +1,428 @@
+  if (instructorIdx !== -1 && instructorsIdx !== -1) {
+    usableFields.splice(instructorIdx, 1);
+  }
+  // If both "prefix" and "prefixes" are present, ignore "prefix"
+  if (prefixIdx !== -1 && prefixesIdx !== -1) {
+    usableFields.splice(prefixIdx, 1);
+  }
+
+  // Parse each row of the CSV as an object
+  let object;
+  let value: string;
+  let regMatch: RegExpMatchArray | null;
+  let roomParts: string[];
+  let names: string[];
+  let nameParts: string[];
+  for (let i = 0; i < data.length; i += 1) {
+    object = data[i];
+
+    // Reset defaults
+    name = "";
+    prefixes = [];
+    number = "";
+    letter = "";
+    studentHours = 0;
+    facultyHours = 0;
+    hourPart = "8";
+    numHourPart = 8;
+    minPart = "00";
+    ampm = "AM";
+    startTime = "8:00 AM";
+    duration = 50;
+    building = "";
+    roomNumber = "";
+    roomCapacity = 30;
+    year = new Date().getFullYear();
+    term = di.Term.Fall;
+    half = di.Half.Full;
+    days = [di.Day.Monday, di.Day.Wednesday, di.Day.Friday];
+    globalMax = 30;
+    localMax = 30;
+    anticipatedSize = 30;
+    instructors = [];
+    comments = "";
+
+    // Iterate through the fields of the CSV, and parse their values for this object
+    for (let j = 0; j < usableFields.length; j += 1) {
+      field = usableFields[j];
+      if (field in object) {
+        value = String(object[field]);
+        switch (field) {
+          case "name": {
+            name = value;
+            break;
+          }
+          case "prefixes":
+          case "prefix": {
+            prefixes = value.replace(" ", "").split(/[;,]/);
+            break;
+          }
+          case "number": {
+            number = value;
+            break;
+          }
+          case "section": {
+            letter = value;
+            break;
+          }
+          case "studentHours": {
+            studentHours = Number(value);
+            break;
+          }
+          case "facultyHours": {
+            facultyHours = Number(value);
+            break;
+          }
+          case "startTimeStr":
+          case "startTime": {
+            // Look for a time
+            regMatch = value.match(timeReg);
+            if (regMatch != null && regMatch.length === 3) {
+              // Get the hour and minute values, store as number and strings
+              [, hourPart, minPart] = regMatch;
+              numHourPart = Number(hourPart);
+
+              // Handle high hour values
+              if (numHourPart > 11) {
+                if (numHourPart > 12) {
+                  // If military time, convert to standard
+                  numHourPart -= 12;
+                  hourPart = String(numHourPart);
+                }
+                // Assume PM when 12:XX or military time
+                ampm = "PM";
+              }
+
+              // If hour is 0, assume military time of 12 AM
+              if (numHourPart === 0) {
+                hourPart = "12";
```

Comment:
If a case is more than just two lines (assignment and break), I would suggest refactoring this code to become a helpful function that the case calls to help with the readability.

Reply:
There are a few other cases where it would be good to abstract the code to its own function.

Reply:
4c83ce3